analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/ObjectCreation.cs

```diff
@@ -29,7 +30,13 @@ internal sealed class ObjectCreation : SimpleProcessor<IObjectCreationOperationW
 
     protected override ProgramState Process(SymbolicContext context, IObjectCreationOperationWrapper operation)
     {
-        if (operation.Type.IsNullableValueType())
+        if (operation.Type.IsAny(EmptyCollectionsShouldNotBeEnumeratedBase.TrackedCollectionTypes)
+            && CollectionCreationConstraint(context.State, operation) is { } constraint)
+        {
+            return context.SetOperationConstraint(constraint)
+                .SetOperationConstraint(operation, ObjectConstraint.NotNull);
```

Comment:
Is setting the `NotNull` constraint as well useful?
It didn't happen before, but I do not think it can be null.

Reply:
Based on the UTs, it seems to be working.

Reply:
I misunderstood what you meant.
You are both correct and wrong:
- It does not work if I remove it
- There is indeed code somewhere that is setting it to `NotNull`. And it's 20 lines down :)

I basically over-wrote the previous behavior by adding this if-arm.
I re-structured the conditionals a bit, I think it makes more sense no